app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/adapter/UserSettingsAdapter.kt

```diff
@@ -1,25 +1,26 @@
 package com.github.sdpteam15.polyevents.model.database.remote.adapter
 
+import com.github.sdpteam15.polyevents.model.database.local.entity.UserSettings
 import com.github.sdpteam15.polyevents.model.database.remote.DatabaseConstant.UserSettingsConstant.*
-import com.github.sdpteam15.polyevents.model.room.UserSettings
 
 /**
  * A class for converting between user setting entities in our code and
  * documents in the Firebase database. Not unlike the conversion to
  * DTO (Data transfer object) concept.
  */
-object UserSettingsAdapter: AdapterInterface<UserSettings> {
-    override fun toDocument(element: UserSettings): HashMap<String, Any?> =
-        hashMapOf(
-                USER_SETTINGS_TRACK_LOCATION.value to element.trackLocation,
-                USER_SETTINGS_SENDING_LOCATION_ON.value to element.isSendingLocationOn,
-                USER_SETTINGS_LOCATION_ID.value to element.locationId
+object UserSettingsAdapter : AdapterInterface<UserSettings> {
+    override fun toDocument(element: UserSettings?): HashMap<String, Any?>? =
+        if (element == null) null
+        else hashMapOf(
+            USER_SETTINGS_TRACK_LOCATION.value to element.trackLocation,
+            USER_SETTINGS_SENDING_LOCATION_ON.value to element.isSendingLocationOn,
+            USER_SETTINGS_LOCATION_ID.value to element.locationId
         )
 
-    override fun fromDocument(document: MutableMap<String, Any?>, id: String): UserSettings =
+    override fun fromDocument(document: Map<String, Any?>, id: String): UserSettings =
```

Comment:
Why did you make the element nullable ? I mean, why should we use the adapter if the element is null.

Reply:
With the local cache an element in the remote db can be invalid so a adapter can return null if its the case